wasmcloud-chat/actors/presence/src/kv.rs

```diff
@@ -0,0 +1,75 @@
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+use std::iter::FromIterator;
+use wapc_guest::prelude::*;
+use wasmcloud_actor_keyvalue as keyvalue;
+use wasmcloudchat_presence_interface::*;
+
+const USER_PRESENCE_EXPIRATION_SECONDS: i32 = 10 * 60; // 10 minutes
+
+pub(crate) fn store_tracking_report(report: PresenceTrackReport) -> HandlerResult<()> {
+    let detail_key = format!("presence:{}", report.user_id);
+    let mut detail = get_user_detail(&report.user_id)?.unwrap_or(UserPresence {
+        user_id: report.user_id.to_string(),
+        channels: vec![],
+    });
+    let mut hs: HashSet<String> = HashSet::from_iter(detail.channels.iter().cloned());
+    hs.insert(report.channel_id);
+
+    detail.channels = hs.into_iter().collect();
+
+    let _ = keyvalue::default().set(
+        detail_key,
+        serde_json::to_string(&detail)?,
+        USER_PRESENCE_EXPIRATION_SECONDS,
+    )?;
+
+    let set_key = "presence:users";
```

Comment:
I think this could be a `const` as it's used in more than one place, and may avoid writing the wrong key later on

Reply:
It should be... not sure why I didn't do that this time.